### PR TITLE
Example Added: Simple File Based Routing Server using Dynamic Import

### PR DESCRIPTION
**Pull request by naishe**
Added a simplest possible example of file based routing like Next.js does. This also shows the usage of dynamic import.
I tried to add as much references and comments for clarity without being too verbose.

Now, send me goodies! :)

### CODE REVIEW

File: examples/http-server-file-router.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  // Check if the module has the method handler.
+  if (module[method]) {
+    // Call the method handler with the request.
+    return module[method](req);
+  }
+
+  // If the method handler is not found, return a 501:Not Imlemented response.
+  return new Response("Method not implmeneted", { status: 501 });
```
Comment (thisisjofrank) [inline examples/http-server-file-router.ts:66]:
```suggestion
  return new Response("Method not implemented", { status: 501 });
```

File: examples/http-server-file-router.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    return new Response("Not found", { status: 404 });
+  }
+
+  // Check if the module has the method handler.
+  if (module[method]) {
+    // Call the method handler with the request.
+    return module[method](req);
+  }
+
+  // If the method handler is not found, return a 501:Not Imlemented response.
```
Comment (thisisjofrank) [inline examples/http-server-file-router.ts:65]:
```suggestion
  // If the method handler is not found, return a 501:Not Implemented response.
```

**Review by thisisjofrank — APPROVED** [review]

**Comment by naishe** [comment]
Thanks @thisisjofrank for merging. Does the doc pages don't work well be generics notations. For example, you could see that the code has the following line:

```ts
async function handler(req: Request): Promise<Response> {
```

However, [the docs page](https://docs.deno.com/examples/http-server-file-router/) renders by removing the angled bracket.

```ts
async function handler(req: Request): Promise {
```

Screenshot:
Screenshot

Is it a known bug? I couldn't find an open reported issue about it.

**Comment by thisisjofrank** [comment]
Oh, interesting, no that's not a known bug, thank you for raising an issue,
I'll take a look at it today!

**Comment by thisisjofrank** [comment]
For now I've moved the example over into tutorials, which fixes one of the problems, will look into the angle brackets issue next
https://github.com/denoland/docs/pull/1061

**Pull request merged**